packages/app/src/utils/tenderSwapHooks.ts

```diff
@@ -61,18 +60,59 @@ export const useCalculateRemoveLiquidityOneToken = (
   return result?.value?.[0] ?? constants.Zero;
 };
 
-export const getExecutionPrice = (receiveAmount: BigNumber, sendAmount: string) => {
-  return receiveAmount.div(hasValue(sendAmount) ? sendAmount : 1);
-};
-
-export const usePriceImpact = (
+export const useSwapPriceImpact = (
   isSendingToken: boolean,
   pool: string,
   inputAmount: string,
   receiveAmount: BigNumber
 ) => {
   const swapContract = new Contract(pool, TenderSwapABI) as TenderSwap;
 
+  const resultVirtualPrice = useCall(
+    pool && {
+      contract: swapContract,
+      method: "getVirtualPrice",
+      args: [],
+    }
+  );
+
+  const virtualPrice = resultVirtualPrice?.value?.[0] ?? constants.One;
+
+  const priceImpact = calculatePriceImpact(
+    utils.parseEther(inputAmount === "" ? "0" : inputAmount),
+    receiveAmount,
+    virtualPrice,
+    !isSendingToken
+  );
+
+  return { priceImpact };
+};
+
+export const calculatePriceImpact = (
+  tokenInputAmount: BigNumber,
+  tokenOutputAmount: BigNumber,
+  virtualPrice: BigNumber,
+  isWithdraw = false
+) => {
+  if (tokenInputAmount.lte(0)) {
+    return constants.Zero;
+  }
+
+  return isWithdraw
+    ? (weiToEthInFloat(tokenOutputAmount) / (weiToEthInFloat(tokenInputAmount) * weiToEthInFloat(virtualPrice)) - 1) *
+        100
+    : ((weiToEthInFloat(virtualPrice) * weiToEthInFloat(tokenOutputAmount)) / weiToEthInFloat(tokenInputAmount) - 1) *
+        100;
```

Comment:
I think percentages should be returned as a decimal representation (i.e. 5% is 0.05) and only be converted to % for display on the front-end in the component itself.